### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by richardeoin**
* Add missing links to examples from docs
* Add a description to more examples
* Update some comments

### CODE REVIEW

**Review by jordens — CHANGES_REQUESTED** [review]
Otherwise LGTM.

File: examples/adc.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! Example of reading a voltage with ADC1
+//! Example of reading a voltage with ADC1 through DMA
```
[the remark this answers is not in the document]
Comment (richardeoin) [reply examples/adc.rs:1]: Thanks for the review, indeed a typo

File: examples/adc_dma.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-//! Example of reading ADC1 data using DMA1/2
+//! Example of reading ADC1 data using DMA1/2 through DMA
```
Comment (jordens) [inline examples/adc_dma.rs:1]: I think this was fine before

```suggestion
//! Example of reading ADC1 data using DMA1
```

**Review by jordens — CHANGES_REQUESTED** [review]

File: examples/adc_dma.rs (hunk 2 above)
Comment (richardeoin) [reply examples/adc_dma.rs]: Thanks, corrected now

**Pull request merged**